x/autopilot/module_ibc.go

```diff
@@ -169,9 +169,31 @@ func (im IBCModule) OnRecvPacket(
 
 	//// At this point, we are officially dealing with an autopilot packet
 
-	// Modify the packet data by replacing the JSON metadata field with a receiver address
-	// to allow the packet to continue down the stack
-	tokenPacketData.Receiver = autopilotMetadata.Receiver
+	// Update the reciever in the packet data so that we can pass the packet down the stack
+	// (since the "receiver" may have technically been a full JSON memo)
+	switch routingInfo := autopilotMetadata.RoutingInfo.(type) {
+	case types.StakeibcPacketMetadata:
+		// For autopilot liquid stake and forward, we'll use a hashed receiver address instead
+		// that will also be used as the sender in the forwarding step
+		// This is to prevent impersonation at downstream zones
+		// We can identify the forwarding step by whether there's a non-empty IBC receiver field
+		if routingInfo.Action == types.LiquidStake && routingInfo.IbcReceiver != "" {
```

Comment:
Is `routingInfo.IbcReceiver != ""` the best way to identify that a packet is a LS&forward packet?

Reply:
this is temporary for now - will replace with a switch on the action after we change the schema